packages/lesswrong/server/sql/SqlFragment.ts

```diff
@@ -54,31 +54,138 @@ export const getResolverCollection = (
   return getCollectionByTypeName(type);
 }
 
+function extractFragmentDefinitions(fragmentSrc: string): Record<string, string> {
+  // Matches on e.g. `fragment PostsList on Post {`
+  const fragmentDeclarations = fragmentSrc.matchAll(/fragment\s+([a-zA-Z0-9-_]+)\s+on\s+([a-zA-Z0-9-_]+)\s*\{/g);
+  const fragments: Record<string, string> = {};
+  for (const match of fragmentDeclarations) {
+    const [declaration, name] = match;
+    const startIndex = fragmentSrc.indexOf(declaration) + declaration.length;
+    let openBraceCount = 1;
+    let traversalIndex = startIndex;
+
+    while (openBraceCount > 0) {
+      const char = fragmentSrc[traversalIndex];
+      if (char === '{') {
+        openBraceCount++;
+      } else if (char === '}') {
+        openBraceCount--;
+      }
+      traversalIndex++;
+    }
+
+    const fragmentText = fragmentSrc.slice(startIndex - declaration.length, traversalIndex);
+    fragments[name] = fragmentText;
+  }
+  return fragments;
+}
+
+// Build a dependency graph and perform topological sort on fragments
+function topologicalSort(fragments: Record<string, string>): [string, string[]][] {
+  const graph: Record<string, string[]> = {};
+  const visited: Record<string, boolean> = {};
+  const temp: Record<string, boolean> = {}; // For cycle detection
+  const result: string[] = [];
+  
+  // Build the dependency graph by finding "...FragmentName" patterns
+  for (const fragmentName in fragments) {
+    const fragmentBody = fragments[fragmentName];
+    const dependencies: string[] = [];
+    
+    // Find fragment references (spread operators)
+    const spreadMatches = fragmentBody.matchAll(/\.\.\.([\w-]+)/g);
```

Comment:
I don't think `-` is legal in graphql fragment names
```suggestion
    const spreadMatches = fragmentBody.matchAll(/\.\.\.([\w]+)/g);
```

Reply:
👍